Don't allow the scimoz plugin element to be focusable, as we always want the
scintilla element to be focused instead.

Index: /Users/toddw/as/komodo-devel_12/mozilla/build/moz1700-ko8.12/mozilla/layout/generic/nsObjectFrame.cpp
--- a/layout/generic/nsObjectFrame.cpp	Thu Oct 18 16:40:12 2012 +1100
+++ b/layout/generic/nsObjectFrame.cpp	Sat Nov 03 17:32:42 2012 -0700
@@ -787,16 +788,23 @@
       mWidget->SetParent(nullptr);
     }
   }
 }
 
 bool
 nsObjectFrame::IsFocusable(int32_t *aTabIndex, bool aWithMouse)
 {
+  // ACTIVESTATE: don't allow scimoz plugin element to become focused.
+  nsAutoString embed_type;
+  if (mContent->GetAttr(kNameSpaceID_None, nsGkAtoms::type, embed_type) &&
+      embed_type.LowerCaseEqualsLiteral("application/x-scimoz-plugin")) {
+    return false;
+  }
+  // ACTIVESTATE: end
   if (aTabIndex)
     *aTabIndex = -1;
   return nsObjectFrameSuper::IsFocusable(aTabIndex, aWithMouse);
 }
 
 bool
 nsObjectFrame::IsHidden(bool aCheckVisibilityStyle) const
 {
